modules/tinymce/src/core/test/ts/browser/fmt/ExpandRangeTest.ts

```diff
@@ -270,6 +270,14 @@ describe('browser.tinymce.core.fmt.ExpandRangeTest', () => {
       const rng = expandRng(editor, [ 0, 0 ], 1, [ 0, 0 ], 2, selectorFormat, false);
       assertRange(editor, rng, [], 0, [], 1);
     });
+
+    it('TINY-10154: should expand over the nested summary content instead of parent details body element', () => {
```

Comment:
Maybe also add a test to the formatter since you could still run the apply H1 on a summary without having the accordion plugin enabled so the change is in core but affects the accordion. Just a simple tests that test this on a regular `<details><summary>text</summary>blah</details>` element.

Reply:
done